### PR TITLE
changed min cmake ver and CMAKE_SOURCE_DIR to PROJECT_SOURCE_DIR

### PR DESCRIPTION
**Pull request by brightprogrammer**
There was a build error when building CSFML stating that it failed to find files in `cmake` folder. Also there were some warnings for cmake version so changed cmake version too. This fixed the build for me.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 
-cmake_minimum_required(VERSION 2.8.3)
+# version 2.8.3 is deprecated so switch to 3.0
```
Comment (eXpl0it3r) [inline CMakeLists.txt:2]: Please remove this comment

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
+cmake_minimum_required(VERSION 3.0)
```
Comment (eXpl0it3r) [inline CMakeLists.txt:3]: Can you make it `3.0.2` so it's inline with SFML 2.5.1

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
Thanks for the change!

Just two small comments 😉

File: CMakeLists.txt (hunk 1 above)
Comment (brightprogrammer) [reply CMakeLists.txt:2]: Sure, will do that

File: CMakeLists.txt (hunk 2 above)
Comment (brightprogrammer) [reply CMakeLists.txt]: ok it's done

**Comment by eXpl0it3r** [comment]
Can you change the following line

https://github.com/SFML/CSFML/blob/317f58fa318450265b18657ce94c7224cae4c870/cmake/Config.cmake#L38

to something like this?

```
if(CMAKE_CXX_COMPILER MATCHES "clang[+][+]" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Not sure if it's the newer CMake version, but seems like it fails to detect AppleClang on macOS now.

----

Additionally, could you squash the commits into one?

**Comment by brightprogrammer** [comment]
I'm sorry, I'm not that experienced with git, I think the squashing didn't go well but I made the requested changes

**Comment by brightprogrammer** [comment]
> I'm sorry, I'm not that experienced with git, I think the squashing didn't go well but I made the requested changes

ok squashing worked, can you please check now? :smile:

**Comment by brightprogrammer** [comment]
is it possibble that the problem is in the machine on which builds are running on?

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
I managed to get it working, by using the code from SFML's 2.6.x branch 😄

**Comment by eXpl0it3r** [comment]
Thanks a lot for this fix! 🙂

**Pull request merged**